subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4006,6 +4016,11 @@ int lwm2m_engine_context_close(struct lwm2m_ctx *client_ctx)
 
 void lwm2m_engine_context_init(struct lwm2m_ctx *client_ctx)
 {
+	messages_clear(client_ctx);
```

Comment:
It's more of a cleanup than initialization, IMO it would fit better inside `lwm2m_engine_context_close`